### PR TITLE
changing firebasemediate and structure of database

### PR DESCRIPTION
**Pull request by levYuval1007**
changing the structure of the database - every roomateUser or apartmentUser have preferences list , every preferences list is not an arraylist (before). Every value will be represented as an individual in that list.Also , changing the code of the firebase mediate so it will work with the new structure of the database.

### CODE REVIEW

File: app/src/main/java/com/example/roome/ChoosingActivity.java [hunk 1]
```diff
@@ -69,12 +72,15 @@ protected void onCreate(Bundle savedInstanceState) {
...
      */
     private void addListenerToFirebaseDbReference(final ProgressBar progressBar) {
-        firebaseDatabaseReference.addListenerForSingleValueEvent(new ValueEventListener() {
+        firebaseDatabaseReference.child("users").child("RoommateSearcherUser").addListenerForSingleValueEvent(new ValueEventListener() {
             @Override
             public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-                FirebaseMediate.setDataSnapshot(dataSnapshot);
-                allApartmentSearcherIds[0] = FirebaseMediate.getAllApartmentSearcherIds();
-                allRoommateSearcherIds[0] = FirebaseMediate.getAllRoommateSearcherIds();
+//                FirebaseMediate.setDataSnapshot(dataSnapshot);
```
Comment (amit-schwartz-u) [inline app/src/main/java/com/example/roome/ChoosingActivity.java]: please delete this

File: app/src/main/java/com/example/roome/FirebaseMediate.java [hunk 2]
```diff
@@ -77,7 +78,7 @@ public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
             @Override
             public void onCancelled(@NonNull DatabaseError databaseError) {
             }
-        });
+        }); //maybe delete the option to update data snapshot if changed
```
Comment (amit-schwartz-u) [inline app/src/main/java/com/example/roome/FirebaseMediate.java:81]: let's leave this. so please remove comment

File: app/src/main/java/com/example/roome/FirebaseMediate.java [hunk 3]
```diff
@@ -217,17 +233,20 @@ public static RoommateSearcherUser getRoommateSearcherUserByUid(String roommateU
      * @return all RoommateSearcherUsers ids from the relevant list in database.
      */
     public static ArrayList<String> getAptPrefList(String list, String aptUid) {
-        GenericTypeIndicator<ArrayList<String>> t = new GenericTypeIndicator<ArrayList<String>>() {
-        };
+//        GenericTypeIndicator<ArrayList<String>> t = new GenericTypeIndicator<ArrayList<String>>() {
```
Comment (amit-schwartz-u) [inline app/src/main/java/com/example/roome/FirebaseMediate.java]: remove comments

File: app/src/main/java/com/example/roome/FirebaseMediate.java [hunk 4]
```diff
@@ -269,6 +289,29 @@ public static RoommateSearcherUser getRoommateSearcherUserByUid(String roommateU
         return allListRoommates;
     }
 
+
+    public static void addToAptPrefList(String list, String aptUid,
```
Comment (amit-schwartz-u) [inline app/src/main/java/com/example/roome/FirebaseMediate.java]: maybe rename this method to addRoommatesIdsToAptPrefList

**Review by amit-schwartz-u — APPROVED** [review]

File: app/src/main/java/com/example/roome/FirebaseMediate.java (hunk 2 above)
Comment (levYuval1007) [reply app/src/main/java/com/example/roome/FirebaseMediate.java:81]: that is a listener of the whole DB

**Pull request merged**